File: Assignment6/Clause.java

class Clause {
  Clause(final int len, final int[] hgs, final int base, final int neck, final int[] xs) {
    this.hgs = hgs; // head+goals pointing to cells in cs
    this.base = base; // heap where this starts
    this.len = len; // length of heap slice
    this.neck = neck; // first after the end of the head
    this.xs = xs; // indexables in head


    System.out.println("Length:"+len);
    System.out.println("Base:"+base);
    System.out.println("Index XS:");
    for(int i=0; i<xs.length;i++)
    System.out.print("  "+xs[i]);
    System.out.println();
    System.out.println("Value of the neck:"+neck);
    System.out.println("Heap: ");
    for(int i=0; i<hgs.length;i++)
        System.out.print("  "+hgs[i]);
  }
  final int len;
  final int[] hgs;
  final int base;
  final int neck;
  final int[] xs;

  }
public class ClauseTest{
	public static void main(String args[])
	{
	    int ab[]; ab=new int[3];int bc[]; bc=new int[3];
        int seed=24;
	    ab=random(23);
		bc=random(24);
		Clause cl=new Clause(seed+10,bc,seed+12,seed+35,ab);
	}
public static int[] random(int seed){
       int[] data;int random;
        data=new int[3];
         for (int i=100,j=0;j<3;i=i*seed,j++){
            random=i*seed+seed*(i-seed);
            data[j]=random;

        }
        return data;

}

}

File: Assignment6/clause.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;
class Clause
{
public:
  int len;
  int * hgs;
  int base;
  int neck;
  int * xs;

  Clause( int len, int hgs[], int base, int neck,int xs[])
  {
    this->hgs=hgs; // head+goals pointing to cells in cs
    this->base=base; // heap where this starts
    this->len=len; // length of heap slice
    this->neck=neck; // first after the end of the head
    this->xs= xs; // indexables in head
    cout<<"Length :";
    cout<<len;
    cout<<"\n";
    cout<<"Base ";
    cout<<base;
    cout<<"\n";
    cout<<"Index xs : \n";
    for(int i=0;i<3;i++)
{
    cout << hgs[i];
    cout<<" ";

}
  cout<<"\n";
  cout<<"Neck : ";
  cout<<neck;
   cout<<"\n";
    cout<<"Heap : \n";
    for(int j=0;j<3;j++)
{
    cout << xs[j];
    cout<<" ";

}

  }
};


int * randomsdq(int seed){


        static int data[3];int random;

        for (int i=100,j=0;j<3;i=i*seed,j++){
            random=i*seed+seed*(i-seed);
            data[j]=random;
        }
        return data;

}
int main()
{
  int *str1;int *str3;
 int dad[3];
  int seed=24;
  str1=randomsdq(23);
  for (int h=0;h<3;h++){
      dad[h]=str1[h];
  }
  str3=randomsdq(24);
Clause b(seed+10,dad,seed+12,seed+35,str3);
  return 0;
}
